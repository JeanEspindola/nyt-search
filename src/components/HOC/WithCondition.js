import React from 'react';

const WithCondition = (condition, render) => Component => props => (
  condition(props)
    ? render
    : <Component {...props} />
);

export default WithCondition;
