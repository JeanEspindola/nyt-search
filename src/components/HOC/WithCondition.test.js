import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import WithCondition from './WithCondition';

describe('WithCondition', () => {
  let Component;
  let WrapperComponent;
  const props = {
    list: [],
  };

  it('renders WithCondition when list is invalid - without crashing', () => {
    WrapperComponent = WithCondition('list')(Component);
    const wrapper = shallow(<WrapperComponent {...props} />);
    expect(wrapper.length).toBeTruthy();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('renders WithCondition when list is valid - without crashing', () => {
    Component = () => (
      <div>Component</div>
    );
    WrapperComponent = WithCondition('list')(Component);
    props.list = [
      { test: 'value' },
    ];

    const wrapper = shallow(<WrapperComponent {...props} />);
    expect(wrapper.length).toBeTruthy();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
