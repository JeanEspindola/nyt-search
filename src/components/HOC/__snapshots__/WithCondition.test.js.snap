// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithCondition renders WithCondition when isLoading false - without crashing 1`] = `
<Component
  list={
    Array [
      Object {
        "test": "value",
      },
    ]
  }
/>
`;

exports[`WithCondition renders WithCondition when isLoading true - without crashing 1`] = `
<p>
  There are no results to display.
</p>
`;

exports[`WithCondition renders WithCondition when list is invalid - without crashing 1`] = `
<p>
  There are no results to display.
</p>
`;

exports[`WithCondition renders WithCondition when list is valid - without crashing 1`] = `
<Component
  list={
    Array [
      Object {
        "test": "value",
      },
    ]
  }
/>
`;
